src/wp-includes/html-api/class-wp-html-tag-processor.php

```diff
@@ -1286,6 +1296,25 @@ private function parse_next_attribute() {
 				$attribute_end,
 				! $has_value
 			);
+
+			return true;
+		}
+
+		/*
+		 * Track the duplicate attributes so if we remove it, all disappear together.
+		 *
+		 * While `$this->duplicated_attributes` could always be stored as an `array()`,
+		 * which would simplify the logic here, storing a `null` and only allocating
+		 * an array when encountering duplicates avoids needless allocations in the
+		 * normative case of parsing tags with no duplicate attributes.
```

Comment:
Not sure I'm reading this correctly, but allocating one empty array (per tag processor) doesn't seem terribly expensive?

Reply:
Maybe ever so slightly clearer, but fine to keep as-is 👍

Reply:
at some point I'll try and make a reasonable benchmark. until then I'll leave it at avoiding allocation. I'm afraid of allocation